### PR TITLE
Go Version Handling

### PR DESCRIPTION
**Pull request by mikeyoung85**
Make sure base module gets all versions assigned to it and limit modules with major release on the end, github.com/libraries/goexample/v3 for example, to have versions only assigned to that major release stream.

### CODE REVIEW

File: spec/models/package_manager/go_spec.rb [hunk 1]
```diff
@@ -46,7 +46,7 @@
   describe "#mapping" do
     it "maps data correctly from pkg.go.dev" do
       VCR.use_cassette("pkg_go_dev") do
-        project = described_class.project("github.com/stretchr/testify")
+        project = described_class.project("github.com/urfave/cli")
```
Comment (jsonperl) [inline spec/models/package_manager/go_spec.rb:49]: This is my fave

Comment (mikeyoung85) [reply spec/models/package_manager/go_spec.rb:49]: `github.com/ourfave/cli`

**Review by jsonperl — APPROVED** [review]
This is perfecto 👌

File: spec/models/package_manager/go_spec.rb [hunk 2]
```diff
@@ -55,6 +55,78 @@
...
+  describe "#update" do
+    it "should queue update for non versioned module" do
+      VCR.use_cassette("pkg_go_dev_v2") do
+        expect(PackageManagerDownloadWorker).to receive(:perform_async).with(described_class.name, "github.com/urfave/cli")
+
+        described_class.update("github.com/urfave/cli/v2")
+      end
+    end
+
+    it "should update both the major release module and base module" do
```
Comment (mikeyoung85) [inline spec/models/package_manager/go_spec.rb:107]: @jsonperl added a spec that does the full run and checks the results in the database

File: app/models/package_manager/go.rb [hunk 3]
```diff
@@ -21,6 +21,8 @@ class Go < Base
       ".svn",
     ].freeze
 
+    VERSION_MODULE_REGEX = /(.+)\/(v\d)/.freeze
```
Comment (tiegz) [inline app/models/package_manager/go.rb:24]: I suppose the day could come when people exceed single digits 😱 

```suggestion
    VERSION_MODULE_REGEX = /(.+)\/(v\d+)/.freeze
```

Comment (jsonperl) [reply app/models/package_manager/go.rb:24]: Oh totally great point! I think we're close already.

**Review by jsonperl — APPROVED** [review]

**Pull request merged**